### PR TITLE
[bhavanatumma/orca-model-cards] - 7b and 13b Orca model cards

### PR DESCRIPTION
**Pull request by BhavanaTumma**
**microsoft-Orca-7b**

**Model:** https://ml.azure.com/model/microsoft-Orca-2-7b:2/details?wsid=/subscriptions/ed2cab61-14cc-4fb3-ac23-d72609214cfd/resourceGroups/sonata-oss-testing-rg/providers/Microsoft.MachineLearningServices/workspaces/sonata-ws-ae&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#overview

**Online Deployment:** https://ml.azure.com/endpoints/realtime/msft-orca-2-7b/test?wsid=/subscriptions/ed2cab61-14cc-4fb3-ac23-d72609214cfd/resourceGroups/sonata-oss-testing-rg/providers/Microsoft.MachineLearningServices/workspaces/sonata-ws-ae&tid=72f988bf-86f1-41af-91ab-2d7cd011db47

**Batch deployment:** https://ml.azure.com/runs/batchjob-9e2574bc-ba4c-cb22-0a81-39895bc896d6?wsid=/subscriptions/b17253fa-f327-42d6-9686-f3e553e24763/resourceGroups/win19/providers/Microsoft.MachineLearningServices/workspaces/11120925024Nov&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#/?graphId=82325d0d-b772-4f2d-be7d-b7e4415c6aaf&label=stoic_chain_2rj9bz61&path=%2Fruns%2Fbatchjob-9e2574bc-ba4c-cb22-0a81-39895bc896d6&runId=batchjob-9e2574bc-ba4c-cb22-0a81-39895bc896d6

image

image

**Conda yaml**

channels:
- conda-forge
dependencies:
- python=3.8.18
- pip<=23.2.1
- pip:
  - mlflow==2.6.0
  - cloudpickle==2.2.1
  - jsonpickle==3.0.2
  - mlflow-skinny==2.6.0
  - azureml-core==1.53.0
  - azureml-mlflow==1.53.0
  - azureml-metrics[all]==0.0.30
  - scikit-learn==1.1.3
  - cryptography==41.0.4
  - python-dateutil==2.8.2
  - datasets==2.14.5
  - soundfile==0.12.1
  - librosa==0.10.1
  - diffusers==0.21.4
  - sentencepiece==0.1.99
  - transformers==4.34.0
  - torch==1.13.1
  - accelerate==0.23.0
  - Pillow==9.4.0
  - azureml-evaluate-mlflow==0.0.30
- pycocotools=2.0.4
name: mlflow-env

**MLModel**

flavors:
  hftransformersv2:
    code: null
    hf_config_class: LlamaConfig
    hf_pretrained_class: LlamaForCausalLM
    hf_tokenizer_class: LlamaTokenizerFast
    huggingface_id: microsoft/Orca-2-7b
    model_data: data
    pytorch_version: 1.13.1
    task_type: text-generation
    transformers_version: 4.34.0
  python_function:
    data: data
    env: conda.yaml
    loader_module: azureml.evaluate.mlflow.hftransformers
    python_version: 3.8.18
metadata:
  base_model_name: microsoft/Orca-2-7b
  base_model_task: text-generation
  is_acft_model: true
  is_finetuned_model: false
mlflow_version: 2.6.0
model_uuid: 8dc2e1503696489183dc559a59d1648e
utc_time_created: '2023-11-30 07:24:34.041230'

**microsoft-Orca-13b**

**Model:** https://ml.azure.com/model/microsoft-Orca-2-13b:1/details?wsid=/subscriptions/ed2cab61-14cc-4fb3-ac23-d72609214cfd/resourceGroups/sonata-oss-testing-rg/providers/Microsoft.MachineLearningServices/workspaces/sonata-ws-ae&tid=72f988bf-86f1-41af-91ab-2d7cd011db47

**Online Deployment:** https://ml.azure.com/endpoints/realtime/msft-orca-2-13b-deploy/test?wsid=/subscriptions/b17253fa-f327-42d6-9686-f3e553e24763/resourceGroups/win19/providers/Microsoft.MachineLearningServices/workspaces/11120925024Nov&tid=72f988bf-86f1-41af-91ab-2d7cd011db47

**Batch deployment:** https://ml.azure.com/runs/batchjob-3d3310ff-8205-0a51-91e3-78a8318dc7fa?wsid=/subscriptions/b17253fa-f327-42d6-9686-f3e553e24763/resourceGroups/win19/providers/Microsoft.MachineLearningServices/workspaces/11120925024Nov&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#/?graphId=675fa9db-9f8a-4c5e-9e81-a669e11bec25&label=sleepy_goat_m8yv10pz&path=%2Fruns%2Fbatchjob-3d3310ff-8205-0a51-91e3-78a8318dc7fa&runId=batchjob-3d3310ff-8205-0a51-91e3-78a8318dc7fa

image

image

**Conda yaml**

channels:
- conda-forge
dependencies:
- python=3.8.18
- pip<=23.2.1
- pip:
  - mlflow==2.6.0
  - cloudpickle==2.2.1
  - jsonpickle==3.0.2
  - mlflow-skinny==2.6.0
  - azureml-core==1.53.0
  - azureml-mlflow==1.53.0
  - azureml-metrics[all]==0.0.30
  - scikit-learn==1.1.3
  - cryptography==41.0.4
  - python-dateutil==2.8.2
  - datasets==2.14.5
  - soundfile==0.12.1
  - librosa==0.10.1
  - diffusers==0.21.4
  - sentencepiece==0.1.99
  - transformers==4.34.0
  - torch==1.13.1
  - accelerate==0.23.0
  - Pillow==9.4.0
  - azureml-evaluate-mlflow==0.0.30
- pycocotools=2.0.4
name: mlflow-env

**MLModel**

flavors:
  hftransformersv2:
    code: null
    hf_config_class: LlamaConfig
    hf_pretrained_class: LlamaForCausalLM
    hf_tokenizer_class: LlamaTokenizerFast
    huggingface_id: microsoft/Orca-2-13b
    model_data: data
    pytorch_version: 1.13.1
    task_type: text-generation
    transformers_version: 4.34.0
  python_function:
    data: data
    env: conda.yaml
    loader_module: azureml.evaluate.mlflow.hftransformers
    python_version: 3.8.18
metadata:
  base_model_name: microsoft/Orca-2-13b
  base_model_task: text-generation
  is_acft_model: true
  is_finetuned_model: false
mlflow_version: 2.6.0
model_uuid: 90b3af0614a4446a8f266b3088e58524
utc_time_created: '2023-11-29 07:52:42.992345'

### CODE REVIEW

File: assets/models/system/microsoft-orca-2-7b/asset.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+extra_config: model.yaml
+spec: spec.yaml
+type: model
+categories: ["Models", "Text Generation"]
```
Comment (skasturi) [inline assets/models/system/microsoft-orca-2-7b/asset.yaml:4]: (nit) add new line at the end of the file?

File: assets/models/system/microsoft-orca-2-7b/spec.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  datasets: Publicly available sources
+  inference-recommended-sku: Standard_NC12s_v3, Standard_NC24s_v3, Standard_NC24rs_v3, Standard_NC8as_T4_v3, Standard_NC16as_T4_v3, Standard_NC64as_T4_v3, Standard_ND40rs_v2, Standard_NC24ads_A100_v4, Standard_NC48ads_A100_v4, Standard_NC96ads_A100_v4, Standard_ND96asr_v4, Standard_ND96amsr_A100_v4
+  languages: en
+tags:
+  Preview: ''
+  inference_compute_allow_list:
+  - Standard_NC12s_v3
+  - Standard_NC24s_v3
+  - Standard_NC24rs_v3
+  - Standard_NC8as_T4_v3
```
Comment (skasturi) [inline assets/models/system/microsoft-orca-2-7b/spec.yaml:15]: Let's add only the ones we have verified on.

File: assets/models/system/microsoft-orca-2-7b/asset.yaml (hunk 1 above)
Comment (BhavanaTumma) [reply assets/models/system/microsoft-orca-2-7b/asset.yaml:4]: Done

**Review by RSavage2 — DISMISSED** [review]

File: assets/models/system/microsoft-orca-2-7b/spec.yaml (hunk 2 above)
Comment (BhavanaTumma) [reply assets/models/system/microsoft-orca-2-7b/spec.yaml]: Done

File: assets/models/system/microsoft-orca-2-13b/description.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
... and money. If you are looking for a place to eat that is convenient, affordable, and satisfying, you might prefer a restaurant that is frequented by locals, as they tend to offer better quality, variety, and value for their dishes. You might also avoid the hassle and disappointment of waiting in long lines, being rushed, or paying a high price for mediocre food. - Enhancing your cultural and social experience. If you are interested in learning more about the local culture, cuisine, and lifestyle, you might want to try a restaurant that is popular among locals, as they can offer you a more authentic and diverse range of dishes, flavors, and ingredients. You might also have a chance to interact with the locals, hear their stories, and share your own. - Avoiding tourist traps and scams. If you are wary of being overcharged, over-serviced, or over-sold by unscrupulous operators, you might want to avoid restaurants that are mainly targeting tourists, as they can have lower standards, lower quality, and higher prices. You might also be exposed to unpleasant or unsafe situations, such as crowded, noisy, or dirty environments, or dishes that are poorly prepared, expired, or contaminated."
```
Comment (novaturient95) [inline assets/models/system/microsoft-orca-2-13b/description.md:71]: the output also contains these tokens. I am guessing we should not include input text for this scenario. Use "return_full_text": false

File: assets/models/system/microsoft-orca-2-13b/description.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    }
+  }
+}
+```
+
+#### Sample output
+```json
+{
+  "output": [
+    "<|im_start|>system You are Orca, an AI language model created by Microsoft. You are a cautious assistant. You carefully follow instructions. You are helpful and harmless and you follow ethical guidelines and promote positive behavior.<|im_end|> <|im_start|>user How can you determine if a restaurant is popular among locals or mainly attracts tourists, and why might this information be useful?<|im_end|> <|im_start|>assistant There are different ways to find out if a restaurant is popular among locals or mainly attracts tourists, depending on the availability and quality of data sources. Some possible methods are: - Checking online reviews and ratings from platforms like Yelp, TripAdvisor, Google Maps, or Zomato, which often indicate the demographics and preferences of the"  ]
```
Comment (novaturient95) [inline assets/models/system/microsoft-orca-2-13b/description.md:97]: same as above

File: assets/models/system/microsoft-orca-2-7b/description.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    }
+  }
+}
+```
+
+#### Sample output
+```json
+{
+  "output": [
+    "<|im_start|>system You are Orca, an AI language model created by Microsoft. You are a cautious assistant. You carefully follow instructions. You are helpful and harmless and you follow ethical guidelines and promote positive behavior.<|im_end|> <|im_start|>user How can you determine if a restaurant is popular among locals or mainly attracts tourists, and why might this information be useful?<|im_end|> <|im_start|>assistant There are different ways to find out if a restaurant is popular among locals or mainly attracts tourists, depending on the availability and quality of data sources. Some possible methods are: - Checking online reviews and ratings from platforms like Yelp, TripAdvisor, Google Maps, or Zomato, which often indicate the demographics and preferences of the"  ]
```
Comment (novaturient95) [inline assets/models/system/microsoft-orca-2-7b/description.md:97]: Please check input and output here as well

File: assets/models/system/microsoft-orca-2-13b/description.md (hunk 3 above)
Comment (BhavanaTumma) [reply assets/models/system/microsoft-orca-2-13b/description.md:71]: in the params?

**Review by skasturi — APPROVED** [review]

File: assets/models/system/microsoft-orca-2-13b/description.md (hunk 3 above)
Comment (novaturient95) [reply assets/models/system/microsoft-orca-2-13b/description.md:71]: Yes, in the parms to prevent input from coming in the output, especially with start end tokens. Otherwise its ok. Non blocker though. Feel free to merge and update in future

**Review by novaturient95 — APPROVED** [review]
